ngraph/core/include/ngraph/op/util/sub_graph_base.hpp

```diff
@@ -5,234 +5,55 @@
 #pragma once
 
 #include <ngraph/op/parameter.hpp>
-#include "ngraph/op/op.hpp"
+#include "ngraph/op/util/multi_subgraph_base.hpp"
 
 namespace ngraph
 {
     namespace op
     {
         namespace util
         {
-            /// \brief Abstract base class for sub-graph based ops, i.e ops that have sub-graph
+            /// \brief Abstract base class for sub-graph based ops, i.e ops that have only one
+            /// sub-graph
             ///
-            class NGRAPH_API SubGraphOp : public Op
+            class NGRAPH_API SubGraphOp : public MultiSubGraphOp
             {
             public:
                 NGRAPH_RTTI_DECLARATION;
-                /// \brief Describes a connection between a SubGraphOp input and the body.
-                class InputDescription
-                {
-                protected:
-                    ///
-                    /// \brief      Constructs a new instance.
-                    ///
-                    /// \param      input_index           Position of the SubGraphOp input
-                    /// \param      body_parameter_index  Body parameter to receive input
-                    ///
-                    InputDescription(uint64_t input_index, uint64_t body_parameter_index);
-                    InputDescription() = default;
-
-                public:
-                    using type_info_t = DiscreteTypeInfo;
-                    virtual ~InputDescription() = default;
-                    virtual std::shared_ptr<InputDescription> copy() const = 0;
-
-                    virtual const type_info_t& get_type_info() const = 0;
-
-                    uint64_t m_input_index{0};
-                    uint64_t m_body_parameter_index{0};
-                };
 
-                ///
-                /// \brief      Describes a body input formed from slices of an input to
-                ///             SubGraphOp.
-                ///
-                class NGRAPH_API SliceInputDescription : public InputDescription
+                virtual std::shared_ptr<Function> get_function() { return m_bodies[0]; };
+                virtual std::shared_ptr<const Function> get_function() const
```

Comment:
```suggestion
                virtual const std::shared_ptr<const Function>& get_function() const
```

Reply:
Do we really need to have both methods?

Reply:
its old code, i dont fix it, in ti and loop sometimes are used theese methods,

Reply:
It happened historically :) I don't remember exactly the whole story. We had get_function() without const qualifier in TensorIterator and SubGraph classes (it was an error), but Gleb G. wanted to use this function in const context, so he added a new function with const qualifier.
I guess we can remain only 
`virtual const std::shared_ptr<const Function>& get_function() const`
But it's not backward compatible change, @ilyachur what do you think should leave both functions or it's better to remain only one?

Reply:
If we don't modify the function I think we can have only one function. In any way, we can decide it in the separate PR.